src/main/java/net/sf/jabref/openoffice/OpenOfficeFileSearch.java

```diff
@@ -0,0 +1,116 @@
+package net.sf.jabref.openoffice;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import net.sf.jabref.logic.journals.JournalAbbreviationRepository;
+
+public class OpenOfficeFileSearch {
+
+    private static final Log LOGGER = LogFactory.getLog(OpenOfficeFileSearch.class);
+
+    private static final String STYLE_FILE_EXTENSION = ".jstyle";
+
+    private final JournalAbbreviationRepository repository;
+
+
+    public OpenOfficeFileSearch(JournalAbbreviationRepository repository) {
+        Objects.requireNonNull(repository);
+        this.repository = repository;
+    }
+
+    /**
+     * Search for Program files directory.
+     * @return the File pointing to the Program files directory, or null if not found.
+     *   Since we are not including a library for Windows integration, this method can't
+     *   find the Program files dir in localized Windows installations.
+     */
+    public List<File> findWindowsProgramFilesDir() {
+        List<String> sourceList = new ArrayList<>();
+        List<File> dirList = new ArrayList<>();
+
+         // 64-bits first
+        String progFiles = System.getenv("ProgramFiles");
+        if (progFiles != null) {
+            sourceList.add(progFiles);
+        }
+
+        // Then 32-bits
+        progFiles = System.getenv("ProgramFiles(x86)");
+        if (progFiles != null) {
+            sourceList.add(progFiles);
+        }
+
+        for (String rootPath : sourceList) {
+            File root = new File(rootPath);
+            File[] dirs = root.listFiles(File::isDirectory);
+            if (dirs != null) {
+                Collections.addAll(dirList, dirs);
+            }
+        }
+        return dirList;
+    }
+
+    /**
+     * If the string dir indicates a file, parse it and add it to the list of styles if
+     * successful. If the string dir indicates a directory, parse all files looking like
+     * style files, and add them. The parameter recurse determines whether we should
+     * recurse into subdirectories.
+     * @param dir the directory or file to handle.
+     * @param recurse true indicates that we should recurse into subdirectories.
+     * @param encoding
+     */
+    public void addStyles(String dir, boolean recurse, Charset encoding, List<OOBibStyle> styles) {
+        File dirF = new File(dir);
+        if (dirF.isDirectory()) {
+            File[] fileArray = dirF.listFiles();
```

Comment:
You could make use of the new JAVA 8 Feature for a Directory as Stream:
 java.nio.file.Files list(Path) returns a Stream<Path> and loads the content successive on demand, only when needed. Also useful for searching specified files:
http://eherrera.net/ocpj8-notes/09-java-file-io-%28NIO.2%29.html#use-stream-api-with-nio-2